utils/src/main/scala/za/co/absa/enceladus/utils/schema/SchemaUtils.scala

```diff
@@ -50,6 +50,27 @@ object SchemaUtils {
     typeHelper(pathTokens, schema)
   }
 
+  def fieldExists(path: String, schema: StructType): Boolean = {
+    def existsHelper(pt: List[String], st: DataType): Boolean = {
+      if (pt.isEmpty) {
+        true
+      } else {
+        st match {
+          case str: StructType => Try {
+            existsHelper(pt.tail, str(pt.head).dataType)
+          }.getOrElse(false)
+          case ArrayType(el: StructType, _) => Try {
```

Comment:
I'm pretty sure this implementation won't work on arrays of arrays or arrays of primitives. It might return `false` if a field actually exists. This is because the implementation relies on the element type of an array to be a StructType.

Reply:
Might be better to have a separate issue/PR to fix this here and in the above `getFieldType ()` method.

Reply:
![image](https://user-images.githubusercontent.com/17467768/64029325-a2bcaf00-cb44-11e9-8d71-87f55c5f3a83.png)

I have an array full of structs and then struct with array of arrays of primitives. I would say it works ok. Or did I miss your concern?

Reply:
Yes. `Array` of `Array` of `Struct`
The function how they were build don't merge into `Array` unless it's `Array` of `Struct`
I needed some changes in the `SchemaUitls` myself. After @yruslan  pointed this issue out, I enhanced my changes So now it can handle nested `Arrays`. And I also added `fieldExist` function and copied your tests. It's all part of  PR #859.
Sorry for messing a little into your code...